src/inhouse.js

```diff
@@ -0,0 +1,250 @@
+            this.discordName = playerDiscordName;
+            this.summonerName = playerSummonerName;
+            this.summonerID;
+            this.rankWeight;
+    
+            this.getSummonerId(this.summonerName).then(response => {
+                this.summonerID = response;
+                //only try to pull more data if we succesfully pulled the ID
+                if(this.summonerID !== null) {
+                    this.getRankWeight(this.summonerID).then(response => {
+                        this.rankWeight = response;
+                        resolve(this);
+                    });
+                }
+            });
+        });
+
+    }
+
+    //retrieves summoner id from summoner name
+    getSummonerId(name) {
+        return new Promise((resolve, reject) => {
+            var _summonerId;
+            let fullUrl = urlSummId + name + urlFinish;
+        
+            //pull data
+            https.get(fullUrl, (resp) => {
+                let data = '';
+
+                // A chunk of data has been recieved.
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                    });
+                // The whole response has been received
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data);
+                        _summonerId = jsonResp.id;
+                        resolve(_summonerId);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+
+                    });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+    }
+    
+    //returns a summoner's solo queue rank weight
+    getRankWeight(summId) {
+        return new Promise((resolve, reject) => {
+            var _rankWeight = 0;
+            var tier;
+            var division;
+    
+            let fullUrl = urlSummRank + summId + urlFinish;
+            //pull json data
+            https.get(fullUrl, (resp) => {
+                let data = '';
+    
+                // A chunk of data has been recieved.
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                  });
+                // The whole response has been received
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data)
+                        var i;
+                        for(i = 0; i < 3; i++) {
+                            if(jsonResp[i] != null) {
+                                if(jsonResp[i].queueType === 'RANKED_SOLO_5x5') {
+                                    tier = jsonResp[i].tier;
+                                    division = jsonResp[i].rank;
+                                    break;
+                                }
+                            }
+                        }
+
+                        //calculate weight
```

Comment:
I'd say move this logic into some utils file called `calculateLeagueRank` or something. I think having this in another file will make this file a lot easier to read through.